docs/new-curriculum/js-core-1/week-1/notes.md

```diff
@@ -0,0 +1,349 @@
+
+:::note
+Important to note that `Math.round` is **implementation opaque**. **Implementation opaque** means we can’t read the set of instructions for `Math.round`. This is because `Math.round` was authored by the developers that created NodeJS.
+:::
+
+### Running scripts
+
+So far we’ve seen how expressions can be evaluated when using the Node REPL. The Node REPL is a very useful tool for evaluating expressions quickly. However, most of the time, we need to write more complex programs that consist of many lines. So we need to write our program in a file. Instead of using the Node REPL, we can use Node to execute the instructions in a file.
+
+We can use the node command to run a JavaScript file. A JavaScript file ends with `.js` - this is known as the file extension.
+
+Let’s suppose we have a file `sums.js` like this:
+
+```js title="sums.js"
+10 + 3;
+10 * 3;
+10 / 3;
+```
+
+In this case, we can write the command `node sums.js`.
+
+This command is an instruction to execute the program written inside index.js. Our program consists of 3 lines, each line an expression. So the machine will start:
+
+`10 + 3;` the computer will evaluate this expression  
+`10 * 3;` the computer will evaluate this expression  
+`10 / 3;` the computer will evaluate this expression
+
+Once the computer evaluates these expressions the execution of the program is complete. But, we’re left with a problem. With the REPL, the user inputs an expression, the computer evaluates it and then prints the result. It then loops back and prompts the user for another input. But now the computer will go and execute each line sequentially until completion.
+
+So this new problem can be expressed as a question:
+
+> #### ❓ Problem
+>
+> "How can we inspect the data in our program during runtime?"
+> In other words: "how can I print values to the terminal when my program is being executed?"
+
+In order to do this, we can use another function.
+
+### Printing to the terminal
+
+In order to print an expression’s result to the terminal we can make use of a function called `console.log`.  
+`console.log` will allow us to print expressions to the console. We also say that we log values to the console. This is very useful for complex programs when we need to inspect the result of certain expressions.
+Let's see how to use `console.log`
+
+In a file called `example.js`, we can write name of the function `console.log`.
+
+```js title="example.js"
+console.log;
+```
+
+However, if we run this file with Node, we won't be able to see anything in the terminal.
+As with `Math.round` we need to use the syntax for calling this function.
+So we can add parentheses after the function name to indicate we're calling this function:
+
+```js title="example.js"
+console.log("hello there!");
+```
+
+Now if we our terminal, we should see the string "hello there!" logged out in the terminal.
+
+### Saving function output
```

Comment:
I think trainees fail to generalise from this to understand that expressions of arbitary complexity can go in any place where an expression can go (e.g. the RHS of an assignment)

Reply:
I agree with this point, but perhaps more concretely this section has no explanation that calling a function is itself an expression, and so just like other expressions, the output can be "stored" as a variable. I do think this is an important thing to be explicit about.

I think this would also clarify the point about `console.log("hello world")` returning `undefined` - this is what the expression evaluates to.